## Importamos librerías y configuraciones

In [24]:
%load_ext autoreload
%autoreload 2

import sys
sys.path.append('../')

from config import *

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


---
---
## Importamos los datos

In [25]:
df = pd.read_pickle('../../datos/tratados/df_employee_data.pkl').reset_index(drop=True)
df.sample()

EnvironmentSatisfaction JobSatisfaction WorkLifeBalance  Age  Attrition  \
3657                     Low            High            Good   55       True   

     BusinessTravel              Department  DistanceFromHome Education  \
3657  Travel_Rarely  Research & Development                19  Bachelor   

        EducationField Gender  JobLevel          JobRole MaritalStatus  \
3657  Technical Degree   Male         1  Human Resources        Single   

      MonthlyIncome  NumCompaniesWorked  PercentSalaryHike  StockOptionLevel  \
3657          46800                6.00                 14                 3   

      TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
3657              24.00                      0              19   

      YearsSinceLastPromotion  YearsWithCurrManager  \
3657                        3                     8   

     Manager_opinion_JobInvolvement Manager_opinion_PerformanceRating  
3657                      Very High                         Excellent

In [26]:
sup_prep.exploracion_datos(df, info=False)

El número de filas es 4410 y el número de columnas es 25

----------

En este conjunto de datos tenemos 2837 valores duplicados

----------

Las principales estadísticas de las variables númericas son:


count     mean      std      min      25%      50%  \
Age                     4410.00    36.92     9.13    18.00    30.00    36.00   
DistanceFromHome        4410.00     9.19     8.11     1.00     2.00     7.00   
JobLevel                4410.00     2.06     1.11     1.00     1.00     2.00   
MonthlyIncome           4410.00 65029.31 47068.89 10090.00 29110.00 49190.00   
NumCompaniesWorked      4391.00     2.69     2.50     0.00     1.00     2.00   
PercentSalaryHike       4410.00    15.21     3.66    11.00    12.00    14.00   
StockOptionLevel        4410.00     0.79     0.85     0.00     0.00     1.00   
TotalWorkingYears       4401.00    11.28     7.78     0.00     6.00    10.00   
TrainingTimesLastYear   4410.00     2.80     1.29     0.00     2.00     3.00   
YearsAtCompany          4410.00     7.01     6.13     0.00     3.00     5.00   
YearsSinceLastPromotion 4410.00     2.19     3.22     0.00     0.00     1.00   
YearsWithCurrManager    4410.00     4.12     3.57     0.00     2.00     3.00   

                             75%       max  
Age                        43.00     60.00  
DistanceFromHome           14.00     29.00  
JobLevel                    3.00      5.00  
MonthlyIncome           83800.00 199990.00  
NumCompaniesWorked          4.00      9.00  
PercentSalaryHike          18.00     25.00  
StockOptionLevel            1.00      3.00  
TotalWorkingYears          15.00     40.00  
TrainingTimesLastYear       3.00      6.00  
YearsAtCompany              9.00     40.00  
YearsSinceLastPromotion     3.00     15.00  
YearsWithCurrManager        7.00     17.00


----------

Las principales estadísticas de las variables categóricas son:


count unique                     top  freq
EnvironmentSatisfaction            4385      4                    High  1350
JobSatisfaction                    4390      4               Very High  1367
WorkLifeBalance                    4372      4                  Better  2660
BusinessTravel                     4410      3           Travel_Rarely  3129
Department                         4410      3  Research & Development  2883
Education                          4410      5                Bachelor  1716
EducationField                     4410      6           Life Sciences  1818
Gender                             4410      2                    Male  2646
JobRole                            4410      9         Sales Executive   978
MaritalStatus                      4410      3                 Married  2019
Manager_opinion_JobInvolvement     4410      4                    High  2604
Manager_opinion_PerformanceRating  4410      2               Excellent  3732


----------

Los columnas con valores nulos y sus porcentajes son: 


%_nulos
WorkLifeBalance             0.86
EnvironmentSatisfaction     0.57
JobSatisfaction             0.45
NumCompaniesWorked          0.43
TotalWorkingYears           0.20

#### Vemos que **tenemos 2837 valores duplicados**. Por lo que vamos a eliminarlos y volver a ver nuestras estadísticas.

In [29]:
df.drop_duplicates(inplace=True)
df.reset_index(drop=True, inplace=True)

sup_prep.exploracion_datos(df, info=False)

El número de filas es 1573 y el número de columnas es 25

----------

En este conjunto de datos tenemos 0 valores duplicados

----------

Las principales estadísticas de las variables númericas son:


count     mean      std      min      25%      50%  \
Age                     1573.00    36.90     9.11    18.00    30.00    36.00   
DistanceFromHome        1573.00     9.16     8.12     1.00     2.00     7.00   
JobLevel                1573.00     2.06     1.11     1.00     1.00     2.00   
MonthlyIncome           1573.00 64979.46 47121.74 10090.00 28990.00 49000.00   
NumCompaniesWorked      1554.00     2.70     2.51     0.00     1.00     2.00   
PercentSalaryHike       1573.00    15.20     3.65    11.00    12.00    14.00   
StockOptionLevel        1573.00     0.79     0.85     0.00     0.00     1.00   
TotalWorkingYears       1564.00    11.25     7.75     0.00     6.00    10.00   
TrainingTimesLastYear   1573.00     2.81     1.29     0.00     2.00     3.00   
YearsAtCompany          1573.00     6.97     6.07     0.00     3.00     5.00   
YearsSinceLastPromotion 1573.00     2.18     3.20     0.00     0.00     1.00   
YearsWithCurrManager    1573.00     4.10     3.57     0.00     2.00     3.00   

                             75%       max  
Age                        43.00     60.00  
DistanceFromHome           14.00     29.00  
JobLevel                    3.00      5.00  
MonthlyIncome           83800.00 199990.00  
NumCompaniesWorked          4.00      9.00  
PercentSalaryHike          18.00     25.00  
StockOptionLevel            1.00      3.00  
TotalWorkingYears          15.00     40.00  
TrainingTimesLastYear       3.00      6.00  
YearsAtCompany              9.00     40.00  
YearsSinceLastPromotion     3.00     15.00  
YearsWithCurrManager        7.00     17.00


----------

Las principales estadísticas de las variables categóricas son:


count unique                     top  freq
EnvironmentSatisfaction            1550      4                    High   482
JobSatisfaction                    1555      4               Very High   490
WorkLifeBalance                    1538      4                  Better   942
BusinessTravel                     1573      3           Travel_Rarely  1117
Department                         1573      3  Research & Development  1030
Education                          1573      5                Bachelor   615
EducationField                     1573      6           Life Sciences   655
Gender                             1573      2                    Male   953
JobRole                            1573      9         Sales Executive   348
MaritalStatus                      1573      3                 Married   720
Manager_opinion_JobInvolvement     1573      4                    High   933
Manager_opinion_PerformanceRating  1573      2               Excellent  1333


----------

Los columnas con valores nulos y sus porcentajes son: 


%_nulos
WorkLifeBalance             2.23
EnvironmentSatisfaction     1.46
NumCompaniesWorked          1.21
JobSatisfaction             1.14
TotalWorkingYears           0.57

#### como tenemos pocos nulos, los vamos a rellenar con el valor 'unknown'

In [31]:
df.fillna('unknown', inplace=True)

df.isna().sum()

EnvironmentSatisfaction              0
JobSatisfaction                      0
WorkLifeBalance                      0
Age                                  0
Attrition                            0
BusinessTravel                       0
Department                           0
DistanceFromHome                     0
Education                            0
EducationField                       0
Gender                               0
JobLevel                             0
JobRole                              0
MaritalStatus                        0
MonthlyIncome                        0
NumCompaniesWorked                   0
PercentSalaryHike                    0
StockOptionLevel                     0
TotalWorkingYears                    0
TrainingTimesLastYear                0
YearsAtCompany                       0
YearsSinceLastPromotion              0
YearsWithCurrManager                 0
Manager_opinion_JobInvolvement       0
Manager_opinion_PerformanceRating    0
dtype: int64